File: SleepStoryBackend/src/main/java/com/sleepstory/backend/domain/entity/UserProfile.java
package com.sleepstory.backend.domain.entity;

import com.sleepstory.backend.domain.valueobject.StoryDuration;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 用户资料实体 - 领域实体
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserProfile {

    private String userId;

    private String sleepTime;

    private StoryDuration preferredDuration;

    private List<String> preferredCategories;

    private Integer streakDays;

    private Integer totalStoriesListened;

    private Integer totalListeningMinutes;

    private LocalDateTime lastListenedAt;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    /**
     * 增加连续天数
     */
    public void incrementStreak() {
        if (this.streakDays == null) {
            this.streakDays = 0;
        }
        this.streakDays++;
        this.updatedAt = LocalDateTime.now();
    }

    /**
     * 重置连续天数
     */
    public void resetStreak() {
        this.streakDays = 0;
        this.updatedAt = LocalDateTime.now();
    }

    /**
     * 记录故事收听
     */
    public void recordListening(int minutes) {
        if (this.totalStoriesListened == null) {
            this.totalStoriesListened = 0;
        }
        if (this.totalListeningMinutes == null) {
            this.totalListeningMinutes = 0;
        }
        this.totalStoriesListened++;
        this.totalListeningMinutes += minutes;
        this.lastListenedAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }
}

File: SleepStoryBackendGo/internal/domain/entity/user_profile.go
package entity

import (
	"time"

	"github.com/sleepstory/backend/internal/domain/valueobject"
)

// UserProfile 用户资料实体
type UserProfile struct {
	UserID                string                   `json:"user_id" gorm:"primaryKey;size:36"`
	SleepTime             string                   `json:"sleep_time" gorm:"column:sleep_time;size:10;default:'22:30'"`
	PreferredDuration     *valueobject.StoryDuration `json:"preferred_duration,omitempty" gorm:"column:preferred_duration;size:20"`
	PreferredCategories   string                   `json:"preferred_categories,omitempty" gorm:"column:preferred_categories;size:200"`
	StreakDays            int                      `json:"streak_days" gorm:"column:streak_days;default:0"`
	TotalStoriesListened  int                      `json:"total_stories_listened" gorm:"column:total_stories_listened;default:0"`
	TotalListeningMinutes int                      `json:"total_listening_minutes" gorm:"column:total_listening_minutes;default:0"`
	LastListenedAt        *time.Time               `json:"last_listened_at,omitempty" gorm:"column:last_listened_at"`
	CreatedAt             time.Time                `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt             time.Time                `json:"updated_at" gorm:"autoUpdateTime"`
}

// TableName 指定表名
func (UserProfile) TableName() string {
	return "user_profiles"
}

// NewUserProfile 创建新用户资料
func NewUserProfile(userID string) *UserProfile {
	medium := valueobject.StoryDurationMedium
	return &UserProfile{
		UserID:                userID,
		SleepTime:             "22:30",
		PreferredDuration:     &medium,
		StreakDays:            0,
		TotalStoriesListened:  0,
		TotalListeningMinutes: 0,
	}
}

// IncrementStreak 增加连续天数
func (p *UserProfile) IncrementStreak() {
	p.StreakDays++
	p.UpdatedAt = time.Now()
}

// ResetStreak 重置连续天数
func (p *UserProfile) ResetStreak() {
	p.StreakDays = 0
	p.UpdatedAt = time.Now()
}

// RecordListening 记录收听
func (p *UserProfile) RecordListening(minutes int) {
	p.TotalStoriesListened++
	p.TotalListeningMinutes += minutes
	now := time.Now()
	p.LastListenedAt = &now
	p.UpdatedAt = now
}
